tbot/main.py

```diff
@@ -57,11 +59,32 @@ def _import_hightlighter() -> typing.Callable[[str], str]:
 # }}}
 
 
+class TbotArgumentParser(argparse.ArgumentParser):
+    def convert_arg_line_to_args(self, arg_line: str) -> List[str]:
+        """
+        Make it possible to use shell variables also in argumentsfiles.
+        """
+        if "$" in arg_line:
+            vl = re.findall(r"(\$\{.*?\})", arg_line)
+            for v in vl:
+                varname = v[2:-1]
+                env = os.getenv(varname)
+
+                if env is None:
+                    raise RuntimeError(f"Could not find Environment variable {varname}")
+
+                # replace in arg_line
+                rp = "${" + varname + "}"
+                arg_line = arg_line.replace(rp, env)
```

Comment:
Actually, there is a much simpler way to do this :)  You don't need to do the string parsing yourself at all!

For expanding undefined variable to nothing:

```python
import string
import os
import collections

environ_default = collections.defaultdict(lambda: "")
environ_default.update(os.environ)

arg_line_expanded = string.Template(arg_line).substitute(environ_default)
```

Or for raising exceptions:

```python
import string
import os
import tbot

try:
    arg_line_expanded = string.Template(arg_line).substitute(os.environ)
except KeyError as e:
    raise tbot.error.TbotException(f"Could not find environment variable: {e.args[0]!r}")
```